### PR TITLE
Drop support for kubernetes < 1.20 for Openstack extension provider

### PR DESCRIPTION
**Pull request by dimitar-kostadinov**
**How to categorize this PR?**

/area open-source
/kind cleanup
/platform openstack

**What this PR does / why we need it**:
Drop support for kubernetes < 1.20 for Openstack extension provider

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6911

**Special notes for your reviewer**:

**Release note**:

```breaking operator
provider-openstack no longer supports Shoots with Кubernetes version < 1.20.
```

### CODE REVIEW

File: pkg/controller/controlplane/valuesprovider.go [hunk 1]
```diff
@@ -347,20 +339,15 @@ func (vp *valuesProvider) GetControlPlaneShootChartValues(
...
 ) (map[string]interface{}, error) {
-	k8sVersionLessThan119, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, "<", openstack.CSIMigrationKubernetesVersion)
-	if err != nil {
-		return nil, err
-	}
-
 	return map[string]interface{}{
 		"volumesnapshots": map[string]interface{}{
-			"enabled": !k8sVersionLessThan119,
+			"enabled": true,
```
Comment (ialidzhikov) [inline pkg/controller/controlplane/valuesprovider.go:350]: Similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/533#discussion_r1050861636.

File: pkg/webhook/controlplane/ensurer.go [hunk 2]
```diff
@@ -328,91 +259,37 @@ var (
...
-		return
-	}
-
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, cloudProviderDiskConfigVolumeMount)
-	// Host certificates are mounted to accommodate OpenStack endpoints that might be served with a certificate
-	// signed by a CA that is not globally trusted.
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, etcSSLVolumeMount)
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, usrShareCACertificatesVolumeMount)
+func ensureKubeAPIServerVolumes(ps *corev1.PodSpec) {
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderDiskConfigVolume.Name)
```
Comment (ialidzhikov) [inline pkg/webhook/controlplane/ensurer.go:273]: Similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/533#discussion_r1050822910

File: charts/internal/cloud-provider-config/values.yaml [hunk 3]
```diff
@@ -1,4 +1,3 @@
-kubernetesVersion: 1.17.0
```
Comment (ialidzhikov) [inline charts/internal/cloud-provider-config/values.yaml:1]: I guess https://github.com/gardener/gardener-extension-provider-openstack/blob/024f99d4dc119437a999beac6c6763e4ce539c45/pkg/controller/controlplane/valuesprovider.go#L476 can be now removed.

File: charts/internal/shoot-system-components/charts/csi-driver-node/templates/_helpers.tpl [hunk 4]
```diff
@@ -11,9 +11,5 @@ cinder.csi.openstack.org
 {{- end -}}
 
 {{- define "csi-driver-node.storageversion" -}}
-{{- if semverCompare "<= 1.18.x" .Values.kubernetesVersion -}}
```
Comment (ialidzhikov) [inline charts/internal/shoot-system-components/charts/csi-driver-node/templates/_helpers.tpl:14]: Can https://github.com/gardener/gardener-extension-provider-openstack/blob/024f99d4dc119437a999beac6c6763e4ce539c45/pkg/controller/controlplane/valuesprovider.go#L755 be now removed?

File: pkg/controller/controlplane/valuesprovider.go [hunk 5]
```diff
@@ -759,37 +728,30 @@ func (vp *valuesProvider) getControlPlaneShootChartValues(
...
-		caBundle = string(caSecret.Data[secretutils.DataKeyCertificateBundle])
+	caSecret, found := secretsReader.Get(caNameControlPlane)
+	if !found {
+		return nil, fmt.Errorf("secret %q not found", caNameControlPlane)
 	}
+	caBundle = string(caSecret.Data[secretutils.DataKeyCertificateBundle])
 
 	csiNodeDriverValues = map[string]interface{}{
-		"enabled":           !k8sVersionLessThan119,
+		"enabled":           true,
```
Comment (ialidzhikov) [inline pkg/controller/controlplane/valuesprovider.go:754]: We could enable the csi-driver-controller chart by default:
https://github.com/gardener/gardener-extension-provider-openstack/blob/1b146ec178cf4f273f41723fd47fa7d36c72bd51/charts/internal/seed-controlplane/values.yaml#L6-L7

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**